src/libsystemd-network/dhcp-option.c

```diff
@@ -41,8 +41,8 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
                 STRV_FOREACH(s, (char **) optval) {
                         size_t len = strlen(*s);
 
-                        if (len > 255)
-                                return -ENAMETOOLONG;
+                        if (len > 255 || len == 0)
+                                return -EINVAL;
```

Comment:
this is an internal API only, right?

Reply:
Sorry, what do you mean?

Reply:
It's a library yes

Reply:
Given the return value is changing, I was just double checking that we are not breaking some public API

Reply:
sd-dhcp-client or sd-dhcp6-client are not public. So, it is safe to change that.
(NetworkManager uses these library, but I hope the change is nicely followed by the devs.)